util/extsort/external_sorter.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extsort
+
+import (
+	"context"
+	"errors"
+)
+
+var (
+	// ErrSorted is returned when the sorter is already sorted, new writers cannot be created.
+	ErrSorted = errors.New("already sorted")
+	// ErrNotSorted is returned when the sorter is not sorted yet, iterators are not ready.
```

Comment:
The comments are more readily, please add more information to error message because the error may be wrapped at different layers, we don't know what's wrong when see "not sorted"

Reply:
This is a  sentinel error like `io.EOF`, maybe it's the caller's responsibility to wrap the error with context.

Reply:
I remove these two errors since it's not necessary to check specific errors from callers. Callers should guarantee that they will not misuse this interface.